### PR TITLE
chore(homepage): separate out api calls to make homepage load more dynamically

### PR DESCRIPTION
**Pull request by pkdotson**
### SUMMARY
This pr is based off feedback from users with slow loading home page from api calls. The initial setup for page load was to batch call for all data on page load. This pr separates out api calls for recent activity and run mine's api calls in parallel on page load. This should allow some data to show faster on homepage without depending on long api requests.

### TEST PLAN
Unit test will be updated as change are made.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/views/CRUD/utils.tsx
Comment (rusackas) [inline superset-frontend/src/views/CRUD/utils.tsx:64]: I wonder if we can type this more specifically.

Comment (pkdotson) [reply superset-frontend/src/views/CRUD/utils.tsx:64]: yep!

File: superset-frontend/src/views/CRUD/utils.tsx
Comment (rusackas) [inline superset-frontend/src/views/CRUD/utils.tsx:94]: Would there be any perceivable benefit to separating/parallelizing these, too?

File: superset-frontend/src/views/CRUD/welcome/Welcome.tsx
Comment (rusackas) [inline superset-frontend/src/views/CRUD/welcome/Welcome.tsx:30]: This is a stupid nit, but I'll throw it out there... `Mine` makes this explicit about what it's used for (which I like) but there's something in me that wants to rename this to `getMyObjs` since it just _sounds_ better. We should probably leave it alone.... but just it does sound weird to my brain, so I'm just just hemming and hawing about it :P

File: superset-frontend/src/views/CRUD/welcome/Welcome.tsx
Comment (rusackas) [inline superset-frontend/src/views/CRUD/welcome/Welcome.tsx:98]: Not a dealbreaker, since it's not the intent of this PR, but it would be nice to add stronger typing to any `any` or `object` types in this area of the code. Maybe track it for a follow-up PR if it's too much feature creep for this one.

File: superset-frontend/src/views/CRUD/welcome/Welcome.tsx
Comment (rusackas) [inline superset-frontend/src/views/CRUD/welcome/Welcome.tsx:119]: Always appreciate comments like this!

**Review by rusackas — APPROVED** [review]
Changes look good to me, so I'll approve. There are some open questions/nits about typing and one more `promise.all` that _might_ be OK to parallelize, so I'll refrain from merging in case you want to address that in this PR. Some parts (e.g. a pass at tighter typing) might be good to open another PR around if you'd rather not tackle it right now.

... last and least, a quick lint-fix looks to be warranted.

File: superset-frontend/src/views/CRUD/welcome/Welcome.tsx
Comment (pkdotson) [reply superset-frontend/src/views/CRUD/welcome/Welcome.tsx:30]: Make sense. I originally want to go with that naming

File: superset-frontend/src/views/CRUD/utils.tsx
Comment (pkdotson) [reply superset-frontend/src/views/CRUD/utils.tsx:94]: these two are need to fill out edited so they should load together when both are finished

File: superset-frontend/src/views/CRUD/welcome/Welcome.tsx
Comment (pkdotson) [reply superset-frontend/src/views/CRUD/welcome/Welcome.tsx]: done

File: superset-frontend/src/views/CRUD/hooks.ts
Comment (ktmud) [inline superset-frontend/src/views/CRUD/hooks.ts:39]:
```suggestion
  defaultLoading?: boolean;
```

Typo, plus this is probably not needed since the state itself doesn't really use `defaultLoading`.

File: superset-frontend/src/views/CRUD/hooks.ts
Comment (ktmud) [inline superset-frontend/src/views/CRUD/hooks.ts:49]: Nit: can we call it `initialLoadingState`?  This function also has too many positional arguments. In general, I'd recommend refactoring any function with >=3 arguments to use `options` object.

File: superset-frontend/src/views/CRUD/utils.tsx
Comment (ktmud) [inline superset-frontend/src/views/CRUD/utils.tsx:36]: Nit: let's make this a shared type with a more specific name (e.g. `CRUDFilterParam`) and put it into `views/CRUD/types.tsx`, as it is obviously used in other components, too.

File: superset-frontend/src/views/CRUD/welcome/ActivityTable.tsx
Comment (ktmud) [inline superset-frontend/src/views/CRUD/welcome/ActivityTable.tsx:28]: Nit: maybe it is just me, but I think `getEditedObjects` or `getUserEditedEntities` would be more readable.

File: superset-frontend/src/views/CRUD/welcome/Welcome.tsx
Comment (ktmud) [reply superset-frontend/src/views/CRUD/welcome/Welcome.tsx:30]: How about `getUserOwnedObjects`?

**Review by ktmud — COMMENTED** [review]
Code LGTM with a couple of small nits (mostly naming suggestions). Didn't have time to test, though.

Happy to stamp once we have design/PM sign-off.

File: superset-frontend/src/views/CRUD/utils.tsx
Comment (pkdotson) [reply superset-frontend/src/views/CRUD/utils.tsx:36]: Makes sense will do!

File: superset-frontend/src/views/CRUD/hooks.ts
Comment (pkdotson) [reply superset-frontend/src/views/CRUD/hooks.ts:49]: Yep, I'll do the refactoring in a follow up ticket.

**Comment by junlincc** [comment]
not sure what's the best way to test, but compared before and after locally. difference in performance is quite noticeable? 
Good to go! 🟢
Before 
https://user-images.githubusercontent.com/67837651/111709439-d7668900-8804-11eb-9ab4-eadee8fd4b87.mov

After
https://user-images.githubusercontent.com/67837651/111709477-e816ff00-8804-11eb-98a2-75d01b7e457d.mov

**Review by junlincc — COMMENTED** [review]
not sure what's the best way to test, but compared before and after locally. difference in performance is quite noticeable? 
Good to go! 🟢
Before 
https://user-images.githubusercontent.com/67837651/111709553-0977eb00-8805-11eb-9abf-0d00d04a2cbb.mov
After
https://user-images.githubusercontent.com/67837651/111709580-172d7080-8805-11eb-81b3-23e0359ea456.mov

**Comment by ktmud** [comment]
https://user-images.githubusercontent.com/335541/111718524-4baa2800-8817-11eb-9939-8624c4134d29.mp4

So it seems once other API is returned, "Recents" was rendered with an empty result before its own API response returns, resulting in a flash of the empty state---this is more noticeable if the `/recent_activity` API is slow or user network is slow in general---I tested this with network speed simulator in Chrome Dev Tools.

Can we get this fixed before merging this PR?

**Comment by pkdotson** [comment]
@ktmud fixed

https://user-images.githubusercontent.com/17326228/111854284-9e4d1800-88db-11eb-933f-2d6af7312b9d.mov

**Comment by ktmud** [comment]
Looks good!

**Review by junlin-qa — APPROVED** [review]
good to go.

-junlin

**Review by ktmud — APPROVED** [review]

**Pull request merged**
